src/FontMetrics.php

```diff
@@ -326,6 +339,84 @@ public function getTextWidth(string $text, $font, float $size, float $wordSpacin
         return $width;
     }
 
+    /**
+     * Maps substrings of text against the provided font list. This is achieved by
+     * parsing each character of the string against the supported glyphs for each
+     * font. Fonts preference is based on the order of the font list.
+     *
+     * Returns an array containing substring information that indicates the
+     * matched font (if any), start index, substring length, and (optionally)
+     * the actual text of the substring.
+     *
+     * @param string $text            The text to map
+     * @param array  $fontFamilies    List of font families to map against
+     * @param string $subtype         The font subtype (italic, bold, etc.)
+     * @param int    $count           The number of matches to return
+     * @param bool   $returnSubstring Should the actual matched text be returned
+     * @return array
+     */
+    public function mapFontsToText(string $text, array $fontFamilies, string $subtype = "normal", int $count = -1, bool $returnSubstring = false): array
+    {
+        $char_mapping = [];
+        $fonts = [];
+
+        foreach ($fontFamilies as $family) {
+            $font = $this->getFont($family, $subtype);
+            if ($font !== null) {
+                $fonts[$family] = $font;
```

Comment:
The array keys (`$family`) are never used, can simply use a list here.

Reply:
I think I was using it at some point, but that logic must've been dropped.